File: src/main/java/me/glaremasters/guilds/commands/admin/manage/CommandAdminTransfer.java
package me.glaremasters.guilds.commands.admin.manage;

import co.aikar.commands.ACFUtil;
import co.aikar.commands.BaseCommand;
import co.aikar.commands.annotation.CommandAlias;
import co.aikar.commands.annotation.CommandCompletion;
import co.aikar.commands.annotation.CommandPermission;
import co.aikar.commands.annotation.Dependency;
import co.aikar.commands.annotation.Description;
import co.aikar.commands.annotation.Single;
import co.aikar.commands.annotation.Subcommand;
import co.aikar.commands.annotation.Syntax;
import co.aikar.commands.annotation.Values;
import me.glaremasters.guilds.exceptions.ExpectationNotMet;
import me.glaremasters.guilds.guild.Guild;
import me.glaremasters.guilds.guild.GuildHandler;
import me.glaremasters.guilds.messages.Messages;
import me.glaremasters.guilds.utils.Constants;
import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;

@CommandAlias("%guilds")
public class CommandAdminTransfer extends BaseCommand {

    @Dependency private GuildHandler guildHandler;

    /**
     * Transfer a guild to another player
     * @param player The player executing the command
     * @param name the name of the guild that's being modified
     * @param newMaster The new master of the guild
     */
    @Subcommand("admin transfer")
    @CommandPermission(Constants.ADMIN_PERM)
    @Description("{@@descriptions.admin-transfer}")
    @CommandCompletion("@guilds @members-admin")
    @Syntax("<guild> <new master>")
    public void execute(Player player, @Values("@guilds") @Single String name, @Values("@members-admin") @Single String newMaster) {

        Guild guild = guildHandler.getGuild(name);

        if (guild == null) {
            ACFUtil.sneaky(new ExpectationNotMet(Messages.ERROR__GUILD_NO_EXIST));
        }

        OfflinePlayer transferPlayer = Bukkit.getOfflinePlayer(newMaster);

        if (transferPlayer == null) {
            ACFUtil.sneaky(new ExpectationNotMet(Messages.ERROR__PLAYER_NOT_FOUND));
        }

        if (guild.getGuildMaster().getUuid().equals(transferPlayer.getUniqueId())) {
            ACFUtil.sneaky(new ExpectationNotMet(Messages.ERROR__TRANSFER_SAME_PERSON));
        }

        guild.transferGuildAdmin(transferPlayer, guildHandler);

        getCurrentCommandIssuer().sendInfo(Messages.TRANSFER__SUCCESS);
    }

}

File: src/main/kotlin/me/glaremasters/guilds/commands/admin/manage/CommandAdminTransfer.kt
/*
 * MIT License
 *
 * Copyright (c) 2023 Glare
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package me.glaremasters.guilds.commands.admin.manage

import ch.jalu.configme.SettingsManager
import co.aikar.commands.BaseCommand
import co.aikar.commands.CommandIssuer
import co.aikar.commands.annotation.CommandAlias
import co.aikar.commands.annotation.CommandCompletion
import co.aikar.commands.annotation.CommandPermission
import co.aikar.commands.annotation.Dependency
import co.aikar.commands.annotation.Description
import co.aikar.commands.annotation.Flags
import co.aikar.commands.annotation.Single
import co.aikar.commands.annotation.Subcommand
import co.aikar.commands.annotation.Syntax
import co.aikar.commands.annotation.Values
import me.glaremasters.guilds.Guilds
import me.glaremasters.guilds.exceptions.ExpectationNotMet
import me.glaremasters.guilds.guild.Guild
import me.glaremasters.guilds.guild.GuildHandler
import me.glaremasters.guilds.messages.Messages
import me.glaremasters.guilds.utils.Constants
import net.milkbowl.vault.permission.Permission
import org.bukkit.Bukkit
import org.bukkit.entity.Player

@CommandAlias("%guilds")
internal class CommandAdminTransfer : BaseCommand() {
    @Dependency lateinit var guilds: Guilds
    @Dependency lateinit var guildHandler: GuildHandler
    @Dependency lateinit var settingsManager: SettingsManager
    @Dependency lateinit var permission: Permission

    @Subcommand("admin transfer")
    @CommandPermission(Constants.ADMIN_PERM)
    @Description("{@@descriptions.admin-transfer}")
    @CommandCompletion("@guilds @members-admin")
    @Syntax("%guild %new-master")
    fun transfer(issuer: CommandIssuer, @Flags("other") @Values("@guilds") guild: Guild, @Values("@members-admin") @Single master: String) {
        val transfer = Bukkit.getOfflinePlayer(master)

        if (guild.guildMaster.uuid == transfer.uniqueId) {
            throw ExpectationNotMet(Messages.ERROR__TRANSFER_SAME_PERSON)
        }

        guild.transferGuildAdmin(transfer, guildHandler, permission)
        currentCommandIssuer.sendInfo(Messages.TRANSFER__SUCCESS)
    }
}
